spec/requests/api/v0/highlights_controller_spec.rb

```diff
@@ -45,24 +43,98 @@
         stub_current_user_uuid(user_uuid)
       end
 
+      context 'bad pagination values' do
+        it 'complains when per_page is too high' do
+          get highlights_path, params: query_params.merge(per_page: 300)
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(response.body).to match(/must be smaller than or equal to 200/)
+        end
+
+        it 'complains when page is too low' do
+          get highlights_path, params: query_params.merge(page: 0)
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(response.body).to match(/must be greater than or equal to 1/)
+        end
+      end
+
       context 'when just one source id is passed in' do
         it('gets the user\'s highlights for that one source') do
           get highlights_path, params: query_params.merge(source_ids: [highlight2.source_id])
           expect(response).to have_http_status(:ok)
           expect(highlights[0][:id]).to eq highlight2.id
         end
+
+        it('returns pagination data') do
+          get highlights_path, params: query_params.merge(source_ids: [highlight2.source_id])
+          expect(response).to have_http_status(:ok)
+          expect(json_response[:meta][:per_page]).to eq 15
+          expect(json_response[:meta][:page]).to eq 1
+          expect(json_response[:meta][:total_count]).to eq 1
+          expect(json_response[:meta][:count]).to eq 1
+        end
       end
 
       context 'when multiple source IDs and scope provided' do
-        it('gets the user\'s highlights in the order of the source IDs and then by order within page') do
-          get highlights_path, params: query_params.merge(
-            source_ids: [highlight3, highlight2, highlight1].map(&:source_id),
-            scope_id: scope_1_id
-          )
-          expect(response).to have_http_status(:ok)
-          expect(highlights.map{|hl| hl[:id]}).to eq(
-            [highlight3, highlight2, highlight1, highlight5, highlight4].map(&:id)
-          )
+        context 'without paging' do
+          it('gets the user\'s highlights in the order of the source IDs and then by order within page') do
+            get highlights_path, params: query_params.merge(
+              source_ids: [highlight3, highlight2, highlight1].map(&:source_id),
+              scope_id: scope_1_id
+            )
+            expect(response).to have_http_status(:ok)
+            expect(highlights.map{|hl| hl[:id]}).to eq(
+              [highlight3, highlight2, highlight1, highlight5, highlight4].map(&:id)
+            )
+          end
+        end
+
+        context 'with paging' do
+          it('gets the user\'s highlights in the order of the source IDs and then by order within page and paginates') do
+            get highlights_path, params: query_params.merge(
```

Comment:
this is one giant test.   perhaps broken up into a test per page?

Reply:
I left as one test since very related but I dried it up with a loop.